### PR TITLE
Add support for Google Style

### PR DESCRIPTION
**Pull request by j6k4m8**
[Google docstrings](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_google.html) are increasingly common. There are definitely [other users](https://github.com/BurntSushi/pdoc/issues/89) who use Google-style strings as well.

I've implemented some very basic parsing for Google-style docstrings in `html-google.mako`. Seems like a project like this could benefit from natively supporting a few common docstring styles, so I'd be willing to generalize this to numpy-style as well and add a flag `--style=google` perhaps.

Looking forward to hearing your thoughts! cc @BurntSushi for discussion as to if we should merge, cc @igordertigor because I thought you might be interested in seeing this. cc @willgray in case you're interested.
## Google Style

```
"""
A short description of the function goes here.

Arguments:
    argname (type): Description.
    secondarg (type : default): Description.

Returns:
    boolean: True on success.
"""
```

### CODE REVIEW

**Comment by j6k4m8** [comment]
Take a look [here](https://github.com/BurntSushi/pdoc/pull/101/files#diff-e5116ef71c494d0e79d056e48749ba23R72) for the actual change. (I moved it to another file to preserve existing behavior.) Essentially just two very dumb regexes.

**Comment by BurntSushi** [comment]
@j6k4m8 Thanks! Two points:

1. A duplicate of `html.mako` with just a few changes is not something I'm personally interested in maintaining.
2. It seems like this doesn't need to be in `pdoc` proper, and someone could instead just modify their own `html.mako` template?

Popping up a level, if we want to support multiple doc string formats, then it might indeed be better to add a CLI flag with a list of supported formats. I am slightly worried about the maintenance burden, however.

**Comment by j6k4m8** [comment]
That's a great point — one of the reasons I added the `don't-merge` was so I could get your input on how best to extend pdoc — I'm pretty unfamiliar with mako. 

Would you be amenable to a variety of parser functions that can be triggered via CLI flags? Looking back at this, I'm not sure why I implemented this way...

**Comment by BurntSushi** [comment]
@j6k4m8 I personally would prefer encouraging one format instead of a divergence of formats. I think the specific thing I'm worried about is supporting the currently very basic and simple linking mechanism. It's a Markdown specific regex that may not apply to other doc string formats. In particular, reST is quite a bit more sophisticated and I'm not particularly interested in doing the work for full integration with it.

If we can find a way to make this really simple, then I could be persuaded to get on board.

**Comment by j6k4m8** [comment]
...alright cleaned it up a ton. You can now call `mark` with a `preprocessor=Preprocessors.GOOGLE`, for instance — and before the text is passed to the `markdown` parser, it's processed via an optional function. 

The output of the preprocessor is still markdown — I'm very much opposed to adding the extra complexity of supporting reST (which sucks anyway) or other formats beyond simply supporting common docstring formats that are nearly markdown anyway.

File: pdoc/templates/html.mako [hunk 1]
```diff
@@ -59,15 +59,35 @@
...
     if not module_list:
       s, _ = re.subn('`[^`]+`', linkify, s)
 
     extensions = []
+
     if use_pygments:
-      extensions = ['markdown.extensions.codehilite(linenums=False)']
+      extensions.append('markdown.extensions.codehilite(linenums=False)')
+
+    if preprocessor:
```
Comment (BurntSushi) [inline pdoc/templates/html.mako]: Where is this set?

File: scripts/pdoc [hunk 2]
```diff
@@ -77,6 +77,9 @@ aa('--http-dir', type=str, default=default_http_dir,
         'server.')
 aa('--http-host', type=str, default='localhost',
    help='The host on which to run the HTTP server.')
+aa('--docstring-style', type=str, default=None,
```
Comment (j6k4m8) [inline scripts/pdoc]: Pass `--docstring-style=google` to parse with a Googly flavor.

**Comment by j6k4m8** [comment]
Run this:
```
pdoc ndio --html --html-dir="pdocs" --overwrite --docstring-style=google
```

...to get this:
image

**Comment by j6k4m8** [comment]
@BurntSushi sorry for the long, drawn out PR comments-section! But I think this is working now — should be pretty clean code, and as far as I could tell, it doesn't get in the way if you want to just run without a `docstring-style` flag. Looking forward to hearing your thoughts!

**Comment by realitix** [comment]
Nice j6k4m8 !
Google documentation is very clean and elegant.

**Comment by igordertigor** [comment]
Cool!

**Comment by j6k4m8** [comment]
Glad you guys are excited about this too!! I haven't looked at this in a while, so it's possible it's not the best approach anymore. But I'll take another look this week

**Comment by peterjc** [comment]
How about calling the new command line argument ``--docformat`` to match ``epydoc``?

Do you think this could be adapted to insert an RST to markdown preprocessor (or other options like epytext), referencing #35 and #111

**Comment by j6k4m8** [comment]
I'm amenable. But ultimately up to @BurntSushi!

I tried to write it as modularly as possible so that you can drop in whatever flavor of parser you like

**Comment by benjamingorman** [comment]
Any idea of if/when this PR will be accepted?

**Comment by cortesi** [comment]
Hi there! This PR conflicts with master. I'll close this for now as part of our general cleanup. Please feel free to rebase this and to resubmit. Pdoc now has a new dev team, and we'll make sure your PR is actioned promptly.

While I'd like to have as few fiddleable knobs in pdoc as posssible, I think having support for some (lightweight and well-defined) alternative comment styles might be plausible. We've decided that complex formats like RST are not on the roadmap.

**Comment by cortesi** [comment]
Please see #153 for a unified discussion ticket about docstring flavors.

**Pull request closed without merging**